prov/efa/src/rdm/efa_rdm_pke_nonreq.h

```diff
@@ -97,6 +67,30 @@ uint64_t *efa_rdm_pke_get_handshake_opt_host_id_ptr(struct efa_rdm_pke *pke)
 	return &handshake_opt_host_id_hdr->host_id;
 }
 
+static inline
+uint32_t efa_rdm_pke_get_handshake_opt_device_version(struct efa_rdm_pke *pke)
+{
+	struct efa_rdm_handshake_hdr *handshake_hdr;
+	struct efa_rdm_handshake_opt_device_version_hdr *device_version_hdr;
+	size_t offset;
+
+	handshake_hdr = efa_rdm_pke_get_handshake_hdr(pke);
+	assert(handshake_hdr->type == EFA_RDM_HANDSHAKE_PKT);
+	assert(handshake_hdr->flags & EFA_RDM_HANDSHAKE_DEVICE_VERSION_HDR);
+
+	offset = sizeof (struct efa_rdm_handshake_hdr)
+		+ ((handshake_hdr->nextra_p3 - 3) * sizeof handshake_hdr->extra_info[0]);
```

Comment:
silly c question:  should it be `sizeof(handshake_hdr->extra_info[0])` ?

Reply:
`sizeof` is a unary operator where the operand can be an expression or a parenthesized data type name. I prefer to stylize `sizeof` expressions to match their semantics, rather than treating `sizeof` as if it's a function/macro.